### PR TITLE
Modify publish post analytics to send more meaningful data

### PR DESCRIPTION
**Pull request by astralbodies**
Resolves: #4081 

Instead of tracking separate events for each qualifier on a published post, attach properties to the original event and drop the four extra events. This means Mixpanel will no longer be incrementing the values and rather a property check will need to be performed to get the counts of published posts with videos for example.

This is targeted for v5.5 as it is resolving a bug with how totals & uniques are handled in Tracks for this particular event.

cc: @meremagee 

Needs Review: @bummytime, @sendhil

### CODE REVIEW

**Comment by astralbodies** [comment]
Also cc: @daniloercoli in case we need to change Android to use the same values.

File: WordPress/Classes/ViewRelated/Post/WPPostViewController.m [hunk 1]
```diff
@@ -1643,23 +1643,12 @@ - (void)logSavePostStats
...
-        
-        if ([self.post hasPhoto]) {
-            [WPAnalytics track:WPAnalyticsStatPublishedPostWithPhoto];
-        }
-        
-        if ([self.post hasVideo]) {
-            [WPAnalytics track:WPAnalyticsStatPublishedPostWithVideo];
-        }
+        properties[WPAnalyticsStatEditorPublishedPostPropertyCategory] = @([self.post hasCategories]);
+        properties[WPAnalyticsStatEditorPublishedPostPropertyPhoto] = @([self.post hasPhoto]);
```
Comment (bummytime) [inline WordPress/Classes/ViewRelated/Post/WPPostViewController.m]: Might be out of scope for this PR, but `[self.post hasPhoto]` is always `0` in my debugger even after adding images/featured image.

File: WordPress/WordPressTest/WPAnalyticsTrackerAutomatticTracksTests.m [hunk 2]
```diff
@@ -37,7 +37,7 @@ - (void)tearDown {
 }
 
 - (void)testVerifyTracksNamesMappings {
-    for (NSUInteger x = 0; x <= WPAnalyticsStatTwoFactorSentSMS; ++x) {
+    for (NSUInteger x = 0; x <= WPAnalyticsStatMaxValue; ++x) {
```
Comment (bummytime) [inline WordPress/WordPressTest/WPAnalyticsTrackerAutomatticTracksTests.m]: :+1:

**Comment by bummytime** [comment]
Besides the `hasPhoto` issue above (which is prolly out of scope). This is a good update, so much cleaner. :-)

:shipit:

**Comment by astralbodies** [comment]
@sendhil awaiting your review since this is going into 5.5

**Comment by sendhil** [comment]
Looks good to me.

**Pull request merged**